### DISCUSSION TITLE
or conditions in graph ql queries

### DISCUSSION THREAD

**Discussion by mihob** [craftcms/cms#12042] [Q&A] [answered]
Is it possible to specify more complex conditions in graph ql queries?

For example, I have entries that represent events. These events have a start date and an optional end date.

Now I want to query all events whose start date is in the future and/or their end date.

Is such a thing even possible and if so, can someone point me in the right direction?

 Thanks in advance!

**Comment by brianjhanson** [comment, marked as answer]
As long as your start & end date fields are date fields, you can query them using the [date fields syntax](https://craftcms.com/docs/4.x/date-time-fields.html#querying-elements-with-date-fields).

In GraphQL the following would get all `event` entries with both a start & end date in the future
```graphql
query {
  entries(section:"events", startDate:"> now", "endDate": "> now") {
    title
  }
}
```

If you need to get more complex, this will get all events with a start date from now until December 1, 2022
```graphql
query {
  entries(section:"events", startDate:["and", "> now", "< 2022-12-01"]) {
    title
  }
}

**Reply by mihob** [reply]
Thank you, but the problem here is that the end date is optional and events can run for several days and should be displayed as long as they are active.

The condition is rather (startDate <= now and endDate = null ) or (endDate != null and endDate <= now) and I don't know how to model this in graphql

**Reply by brianjhanson** [reply]
Makes sense. There's not a way to combine date queries from multiple fields into single set of entries (criteria is always treated as an `and` relationship), but you can use graphql aliases to get two different sets of events and combine them on the front-end. 

```graphql
query {
  currentEvents: entries(
    section: "events"
    startDate: "<= now"
    endDate: ":empty:"
  ) {
    title
  }
  pastEvents: entries(section: "events", endDate: ["or", ":empty:", "<= now"]) {
    title
  }
}
```

**Reply by mihob** [reply]
Yes, but the data is paginated. Therefore, this approach is rather unsuitable. 

But thanks for the help, I'll think about how to solve this best.

**Reply by brianjhanson** [reply]
You can also check out [#9029](https://github.com/craftcms/cms/discussions/9029) and [#9003](https://github.com/craftcms/cms/discussions/9003) for more discussion on this kind of functionality. I'm sure there are more discussion posts around it as well.
